mitiq/pec/representations/tests/test_optimal.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from pytest import mark, raises
+import numpy as np
+from itertools import product
+
+from cirq import (
+    LineQubit,
+    I,
+    X,
+    Y,
+    Z,
+    H,
+    CNOT,
+    CZ,
+    Circuit,
+    DepolarizingChannel,
+    channel,
+    AmplitudeDampingChannel,
+    ResetChannel,
+    reset,
+)
+
+from mitiq.pec.representations.optimal import (
+    minimize_one_norm,
+    find_optimal_representation,
+)
+
+from mitiq.pec.representations.depolarizing import (
+    represent_operation_with_local_depolarizing_noise,
+)
+
+from mitiq.pec.representations.damping import (
+    _represent_operation_with_amplitude_damping_noise,
+)
+
+
+from mitiq.pec.channels import (
+    _operation_to_choi,
+    _circuit_to_choi,
+    global_depolarizing_kraus,
+    amplitude_damping_kraus,
+    kraus_to_super,
+    kraus_to_choi,
+    choi_to_super,
+)
+
+from mitiq.pec.types import OperationRepresentation, NoisyBasis, NoisyOperation
+from mitiq.conversions import convert_from_mitiq
+
+from mitiq.utils import _equal
+
+
+def _equivalent_representations(
```

Comment:
I think we should move this to `OperationRepresentation.__eq__`. Thoughts?

Reply:
Makes sense. Added here bfab2cd